autoPyTorch/pipeline/components/setup/traditional_ml/traditional_learner/learners.py

```diff
@@ -0,0 +1,345 @@
+            self.model = LGBMRegressor(**self.config, random_state=self.random_state)
+        else:
+            self.num_classes = len(np.unique(y_train)) if len(np.unique(y_train)) != 2 else 1  # this fixes a bug
+            self.config["num_class"] = self.num_classes
+
+            self.model = LGBMClassifier(**self.config, random_state=self.random_state)
+
+    def _fit(self, X_train: np.ndarray,
+             y_train: np.ndarray,
+             X_val: np.ndarray,
+             y_val: np.ndarray
+             ) -> None:
+        assert self.model is not None, "No model found. Can't fit without preparing the model"
+        self.model.fit(X_train, y_train, eval_set=[(X_val, y_val)])
+
+    def predict(self, X_test: np.ndarray,
+                predict_proba: bool = False,
+                preprocess: bool = True) -> np.ndarray:
+        assert self.model is not None, "No model found. Can't " \
+                                       "predict before fitting. " \
+                                       "Call fit before predicting"
+        if preprocess:
+            X_test = self._preprocess(X_test)
+
+        if predict_proba:
+            if not self.is_classification:
+                raise ValueError("Can't predict probabilities for a regressor")
+            y_pred_proba = self.model.predict_proba(X_test)
+            if self.num_classes == 2:
+                y_pred_proba = y_pred_proba.transpose()[0:len(X_test)]
+            return y_pred_proba
+
+        y_pred = self.model.predict(X_test)
+        return y_pred
+
+    @staticmethod
+    def get_properties(dataset_properties: Optional[Dict[str, Any]] = None) -> Dict[str, str]:
+        return {
+            'shortname': 'LGBMLearner',
+            'name': 'Light Gradient Boosting Machine Learner',
+        }
+
+
+class CatboostModel(BaseTraditionalLearner):
+
+    def __init__(self,
+                 task_type: str,
+                 output_type: str,
+                 logger_port: int = logging.handlers.DEFAULT_TCP_LOGGING_PORT,
+                 random_state: Optional[np.random.RandomState] = None):
+        super(CatboostModel, self).__init__(name="catboost",
+                                            logger_port=logger_port,
+                                            random_state=random_state,
+                                            task_type=task_type,
+                                            output_type=output_type)
+        self.config["train_dir"] = tempfile.gettempdir()
+
+    def _prepare_model(self,
+                       X_train: np.ndarray,
+                       y_train: np.ndarray
+                       ) -> None:
+        if not self.is_classification:
+            self.config['eval_metric'] = 'R2'
+            # CatBoost Cannot handle a random state object, just the seed
+            self.model = CatBoostRegressor(**self.config, random_state=self.random_state.get_state()[1][0])
+        else:
+            self.config['eval_metric'] = 'Accuracy'
```

Comment:
I Think we should optimize the metric provided by the user if supported by catboost?

Reply:
sure